### PR TITLE
<Added link> Added python and css resources

### LINKED ISSUE #415: [Add] <Go language>
### URL

https://go.dev/learn/

### Description of the above provided site

This webiste teaches step by step with proper explanation.

### Category

Resources

### Sub Category

languages

### PR DESCRIPTION
**Pull request by jayk-gupta**
closes #415 
## Changes proposed
Added resources to the languages section: added python and css resources

## Screenshots
python
css

### CODE REVIEW

**Review by rupali-codes — CHANGES_REQUESTED** [review]
Hey Jay, category should be `Languages`, also it's not showing in the panel

File: database/languages/python.json [hunk 1]
```diff
@@ -1,3 +1,16 @@
 [
-   
+       {
+         "name": "Learn python org",
+         "description": "This website teaches Python in an interactive way by providing in browser code editor with examples.",
```
Comment (CBID2) [inline database/languages/python.json:4]:
```suggestion
         "description": "This website teaches Python in an interactive way by providing a browser code editor with examples.",
```

File: database/languages/python.json [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+         "url": "https://www.learnpython.org/",
+         "category": "Other",
+         "subcategory": "Python"
+       },
+       {
+         "name": "W3schools",
+         "description": "This website teaches Python in an concise and interactive way by providing in browser code editor with examples.",
```
Comment (CBID2) [inline database/languages/python.json:11]:
```suggestion
         "description": "This website teaches Python in an concise and interactive way by providing a browser code editor with examples.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @jayk-gupta! I have a couple of edits on the descriptions of the sources you picked. Other than that, great work! 😊👍🏾

**Comment by jayk-gupta** [comment]
> Hey Jay, category should be `Languages`, also it's not showing in the panel

Updated category.
Not showing means?

**Review by ujjawaltyagii — COMMENTED** [review]
- Resources are not visible on panel, kindly re-check and test the code on your local system too.
- Also there is no such sub-category as ``CSS`` in ``languages`` category so add that sub-category first before adding a different resource. 

Let me know if you are facing any issue in adding new sub-category I will help you or I will directly add that for you.

image

**Review by rupali-codes — APPROVED** [review]

**Review by rupali-codes — CHANGES_REQUESTED** [review]
We already have a category named _css_ in youtube category, can we name it something else?

**Comment by jayk-gupta** [comment]
> * Resources are not visible on panel, kindly re-check and test the code on your local system too.
> * Also there is no such sub-category as `CSS` in `languages` category so add that sub-category first before adding a different resource.
> 
> Let me know if you are facing any issue in adding new sub-category I will help you or I will directly add that for you.
> 
> image

okay, I will let you know.

**Comment by jayk-gupta** [comment]
> We already have a category named _css_ in youtube category, can we name it something else?

That's youtube channel for css and these are websites, so what can we name it then?

File: database/languages/css.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[
+  {
+    "name": "Web dev",
+        "description": "This website teaches CSS in with illustrations and examples which you can edit on CodePen.",
```
Comment (CBID2) [inline database/languages/css.json:4]:
```suggestion
        "description": "This website teaches CSS with illustrations and examples that you can edit on CodePen.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @jayk-gupta! 😊 I just left you a suggestion on how you can improve of the descriptions. Keep it up! 😁 👍🏾

**Review by rupali-codes — COMMENTED** [review]
@jayk-gupta CSS is not a language, better we can add this in _frontend_ category, but since we already have a subcategory it would lead to confusion to have 2 difference subcategories with same name. I guess we should call it _cascading style sheet_.

File: database/languages/css.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        "url": "https://web.dev/learn/css/",
+        "category": "languages",
+        "subcategory": "CSS"
+  },
+  {
+    "name": "learn html css online",
+    "description": "This website teaches CSS in an interactive way by providing code editor and theory side by side.",
+    "url": "https://learnhtmlcss.online/",
+    "category": "languages",
```
Comment (CBID2) [inline database/languages/css.json:13]:
```suggestion
    "category": "frontend",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @jayk-gupta! I helped you created the suggestion that @rupali-codes made.

**Comment by jayk-gupta** [comment]
> Hey Jay, category should be `Languages`, also it's not showing in the panel

I tried adding new json file in the languages subcategory, but it's not showing in the site idk why?
What else do I need to do so that it shows on the site?

**Comment by jayk-gupta** [comment]
> * Resources are not visible on panel, kindly re-check and test the code on your local system too.
> * Also there is no such sub-category as `CSS` in `languages` category so add that sub-category first before adding a different resource.
> 
> Let me know if you are facing any issue in adding new sub-category I will help you or I will directly add that for you.
> 
> image

Yeah, adding new json file but it is not showing on the site?

**Comment by rupali-codes** [comment]
> > Hey Jay, category should be `Languages`, also it's not showing in the panel
> 
> I tried adding new json file in the languages subcategory, but it's not showing in the site idk why? What else do I need to do so that it shows on the site?

You will also have to update `database/index.ts` and `database/data.ts` files accordingly.

**Comment by jayk-gupta** [comment]
> > > Hey Jay, category should be `Languages`, also it's not showing in the panel
> > 
> > 
> > I tried adding new json file in the languages subcategory, but it's not showing in the site idk why? What else do I need to do so that it shows on the site?
> 
> You will also have to update `database/index.ts` and `database/data.ts` files accordingly.

ohh, ohk I will update there also then.
Thanks!

**Comment by jayk-gupta** [comment]
> > > Hey Jay, category should be `Languages`, also it's not showing in the panel
> > 
> > 
> > I tried adding new json file in the languages subcategory, but it's not showing in the site idk why? What else do I need to do so that it shows on the site?
> 
> You will also have to update `database/index.ts` and `database/data.ts` files accordingly.

Updated the files as you said.
In python in the language subcategory the coming soon needs to be removed.

File: database/languages/golang.json [hunk 5]
```diff
@@ -1,3 +1,9 @@
 [
-    
-]
\ No newline at end of file
+   {
+        "name": "Go dev",
+        "description": "This webiste teaches step by step with proper explanation.",
+        "url": "   https://go.dev/learn/",
```
Comment (rupali-codes) [inline database/languages/golang.json:5]:
```suggestion
        "url":  "https://go.dev/learn/",
```

File: database/languages/java.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+[
+    {
+        "name": "Learn Java online",
+        "description": "It is a website to learn java with good examples and excercises.",
+        "url": "https://www.learnjavaonline.org/",
+        "category": "Other",
```
Comment (rupali-codes) [inline database/languages/java.json:6]:
```suggestion
        "category": "languages",

```

File: database/languages/golang.json [hunk 7]
```diff
@@ -1,3 +1,9 @@
...
+        "category": "Other",
```
Comment (rupali-codes) [inline database/languages/golang.json:6]:
```suggestion
        "category": "languages",
```

**Comment by rupali-codes** [comment]
@jayk-gupta could you please resolve the conversations?

**Comment by rupali-codes** [comment]
It's not showing the languages you have added as well as resources in python

File: database/languages/python.json
Comment (rupali-codes) [inline database/languages/python.json:1]: P of Python should be lowercase (_python_)

**Comment by jayk-gupta** [comment]
> Also, not showing java in the languages category
I have made the changes, now it is showing.
Will add more resources soon.
And Sorry, I just forgot about it.

**Review by rupali-codes — APPROVED** [review]
Looks perfect to me, thanks 😄

**Comment by jayk-gupta** [comment]
> Looks perfect to me, thanks 😄

You're welcome! 😊
Will add more resources in 2-3 days in languages category.

File: database/languages/java.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
+[
+    {
+        "name": "Learn Java online",
+        "description": "It is a website to learn java with good examples and excercises.",
```
Comment (CBID2) [inline database/languages/java.json:4]:
```suggestion
        "description": "This website provides good examples and exercises to learn Java.",
```

File: database/languages/golang.json [hunk 9]
```diff
@@ -1,3 +1,9 @@
 [
-    
-]
\ No newline at end of file
+   {
+        "name": "Go dev",
+        "description": "This webiste teaches step by step with proper explanation.",
```
Comment (CBID2) [inline database/languages/golang.json:4]:
```suggestion
        "description": "This webiste teaches Golang with detailed explanations and step by step directions.",
```

File: database/languages/python.json [hunk 10]
```diff
@@ -1,3 +1,16 @@
 [
-   
-]
\ No newline at end of file
+  {
+    "name": "Learn Python ",
+    "description": "This website teaches Python in an interactive way. It has examples, in browser code editor and excercises.",
```
Comment (CBID2) [inline database/languages/python.json:4]:
```suggestion
    "description": "This website teaches Python in an interactive way. It has an online code editor where you go do the exercises alongside the lesson.",
```

File: database/languages/python.json [hunk 11]
```diff
@@ -1,3 +1,16 @@
...
+    "url": "https://www.learnpython.org/",
+    "category": "languages",
+    "subcategory": "python"
+  },
+  {
+    "name": "Programiz",
+    "description": "This website teaches Python in a concise way with examples and also has in browser videos.",
```
Comment (CBID2) [inline database/languages/python.json:11]:
```suggestion
    "description": "This website teaches Python in a concise way with videos and examples.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Well done so far @jayk-gupta! :) I just made some suggestions to improve your hair descriptions.

**Comment by jayk-gupta** [comment]
> Well done so far @jayk-gupta! :) I just made some suggestions to improve your hair descriptions.

Committed all requested changes!

**Comment by rupali-codes** [comment]
@jayk-gupta why did you close the PR?

**Comment by jayk-gupta** [comment]
By mistake I closed it!

**Comment by rupali-codes** [comment]
> By mistake I closed it!

can u reopen it?

**Comment by jayk-gupta** [comment]
I will created a new PR!
May be I have also discarded the changes in the forked repo.

**Comment by rupali-codes** [comment]
> I will created a new PR! May be I have also discarded the changes in the forked repo.

Ohh, okay, awaiting :)

**Pull request closed without merging**